### PR TITLE
bpo-22367: Add test_lockf

### PR DESCRIPTION
**Pull request by corona10**
Before working on [bpo-22367](https://bugs.python.org/issue22367), I 'd like to add unit test for fcntl.lockf

https://bugs.python.org/issue22367

### CODE REVIEW

File: Lib/test/test_fcntl.py [hunk 1]
```diff
@@ -138,6 +137,34 @@ def test_flock(self):
         self.assertRaises(ValueError, fcntl.flock, -1, fcntl.LOCK_SH)
         self.assertRaises(TypeError, fcntl.flock, 'spam', fcntl.LOCK_SH)
 
+    def test_lockf(self):
+        from multiprocessing import Process
+        self.f = open(TESTFN, 'wb+')
+        self.f.write(b'testpython')
+        ex_cmd = fcntl.LOCK_EX | fcntl.LOCK_NB
+        sh_cmd = fcntl.LOCK_SH | fcntl.LOCK_NB
+        def try_lock_f_on_other_process(cmd):
```
Comment (serhiy-storchaka) [inline Lib/test/test_fcntl.py]: Would not be simpler to use two different functions for LOCK_EX and LOCK_SH? Or maybe even split the test on two?

File: Lib/test/test_fcntl.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 fcntl = import_module('fcntl')
 
 
-# TODO - Write tests for flock() and lockf().
```
Comment (serhiy-storchaka) [inline Lib/test/test_fcntl.py:15]: Is the part about flock() still relevant?

Comment (corona10) [reply Lib/test/test_fcntl.py:15]: https://github.com/python/cpython/blob/master/Lib/test/test_fcntl.py#L127

test_flock already exists.
But if the test is not enough, the comment should be remain

**Comment by corona10** [comment]
@serhiy-storchaka
Updated!

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**